### ISSUE TITLE
[FEATURE] Support device (re)hydration

### ISSUE THREAD

**Issue by menturion** [ajbura/cinny#190]
Support device (re)hydration to overcome the issue that encrypted messages cannot be read if the receiving user was offline while the sender sent messages.

See e.g. ...
https://github.com/vector-im/element-web/issues/2996
https://github.com/vector-im/element-web/issues/3825

I know that this feature has not been released officially/finally but device (re)hydration is already implemented in the matrix-react-sdk.

**Comment by notramo** [comment]
Being offline doesn't cause undecryptable messages.

Only if all devices are logged out.

**Comment by menturion** [comment]
That's not correct. If all clients are disconnected from network (i.e. being totally offline) you are not able to decrypt a message you received during this time.

**Comment by notramo** [comment]
Then it's a bug in Cinny. Hydrated devices are not for this. Please open another issue regarding this.

**Comment by kfiven** [comment]
> Then it's a bug in Cinny.

Has not happened to me. @menturion you surely misunderstood the dehydrated device usage. [MSC says](https://github.com/uhoreg/matrix-doc/blob/dehydration/proposals/2697-device-dehydration.md) "_when a user has **no logged-in devices** when a message is sent, they are unable to receive incoming megolm sessions_."

**Comment by menturion** [comment]
@kfiven 

Many thanks for your message! 
Right, this is what I meant. We can change any above description text if this is misleading. 

Device (re)hydration should be supported.

**Comment by akoranii** [comment]
Any progress on this?
How to contribute to fix this issue?
@ajbura @kfiven @menturion

**Comment by Leonscreative** [comment]
Hi, any progress with that? It will be great to resolve it asap because is blocking the User experience. Thanks

**Comment by sceboucher** [comment]
Following.
